src/runtime.rs

```diff
@@ -457,6 +531,37 @@ mod tests {
     }
 
     #[test]
+    fn test_protocol() {
+        let proto = test_utils::custom_protocol();
+        assert!(proto.name() == "CustomProtocol");
+        let class = test_utils::custom_class();
+        assert!(class.conforms_to_protocol(proto));
+    }
+
+    #[test]
+    fn test_protocol_method() {
+        let class = test_utils::custom_class();
+        let result: i32 = unsafe {
+            msg_send![class, addNumber:1 toNumber:2]
+        };
+        assert_eq!(result, 3);
+    }
+
+    #[test]
+    fn test_subprotocols() {
+        let sub_proto = test_utils::custom_subprotocol();
+        let super_proto = test_utils::custom_protocol();
+        assert!(sub_proto.conforms_to_protocol(super_proto));
+    }
+
+    #[test]
+    #[cfg_attr(target_os = "linux", ignore)]
+    fn test_protocols() {
+        let protocols = Protocol::protocols();
```

Comment:
If we call `custom_protocol` before here, can this test be run on Linux? Since that should guarantee at least one protocol is registered with the runtime